### PR TITLE
Add support for MapTiler SDK

### PR DESCRIPTION
**Pull request by petr-hajek**
## Description

As a user of MapTiler SDK & MapTiler maps I'd like an easy way how to compare also MapTiler maps.
With this PR I can add MapTiler API key to the config (similar to MapBox). I can also use MapTiler SDK as a renderer (it's almost the same like MapLibre).

I removed the duplicate code in Map.svelte

## QA steps

Probably some basic regression of MapLibre & MapBox but should not be affected

### Create the PR

- [x] Fill out PR template
- [x] Make sure you've added a CHANGELOG entry under "Unreleased"
- [x] Request a review
- [ ] Make any requested changes and get approval

### After approval

- [ ] Merge any changes from `main` into your branch, resolve any conflicts
- [ ] Squash & merge changes into `main`
- [ ] Delete branch

### CODE REVIEW

**Comment by ebrelsford** [comment]
@petr-hajek Thanks for this! We'll be taking a look at this soon.

**Comment by petr-hajek** [comment]
It's my first PR into this project and for full transparency, MapTiler is my client so I am open to any feedback. Thanks a lot. Anyway,  this is a great tool!

File: src/fetch-url.js [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+  let nextUrl = new URL(url);
   if (urlIsMapbox) {
-    nextUrl = normalizeMapboxUrl(url, mapboxGlAccessToken);
+    nextUrl = normalizeMapboxUrl(nextUrl.toString(), mapboxGlAccessToken);
+  } else if (
+    maptilerApiKey &&
+    nextUrl.host === 'api.maptiler.com' &&
+    !nextUrl.searchParams.has('key')
+  ) {
+    nextUrl.searchParams.append('key', maptilerApiKey);
```
Comment (aparlato) [inline src/fetch-url.js:19]: I'm seeing Maptiler URLs fail in other renderers (Mapbox and Maplibre).

I think it's because here you're following the same pattern we use for a mapbox URL and as I think on this, I wonder if we need a change in this PR for Maptiler and a followup for Mapbox URLs.

Basically we haven't run into this with Mapbox URLs (with the mapbox protocol `mapbox://`) because we've considered them only valid via the Mapbox renderer despite us normalizing the URL here. This is because they typically also reference additional Mapbox protocol URLs for sprites and glyphs so we didn't consider this a bug (You'll see Mapbox styles using the protocol fail in Maplibre for example).

Maptiler (correct me if I'm wrong) is different in that it does not use a custom protocol inside the style and instead creates a valid url with the key present when requesting the style url with that key: 
```
"glyphs": "[https://api.maptiler.com/fonts/{fontstack}/{range}.pbf?key=fBwWvQI476rYQ6q9LrJf](https://api.maptiler.com/fonts/%7Bfontstack%7D/%7Brange%7D.pbf?key=fBwWvQI476rYQ6q9LrJf)"
```

So basically, we should be able to request the style with the new `maptilerApiKey` param while rendering with Maplibre, but we fail [here](https://github.com/stamen/maperture/blob/main/src/components/GlMap.svelte#L68) because it looks at the URL (without accounting for the API key) before the style requested from the normalized the URL.

**tl;dr**
- I think [this line](https://github.com/stamen/maperture/blob/main/src/components/GlMap.svelte#L68) should privilege `style` over `url` (I don't think there's a side effect of this?) like this:
  ```
  map.setStyle(style || url);
  ```
- As a followup, we should consider normalizing the URLs using the Mapbox protocol _within_ a stylesheet to allow it to be rendered outside of the Mapbox renderer.


Let me know if this does not sound accurate!

**Review by aparlato — CHANGES_REQUESTED** [review]
Looking good here!

I left a long explainer for this, but basically I think we may need to switch the order of our `||` conditional [here](https://github.com/maptiler/maperture/blob/main/src/components/GlMap.svelte#L72) to be `style || url` as opposed to how it is now.

We'll also want to QA that change and make sure there's no weird side effects.

cc @ebrelsford

File: src/fetch-url.js (hunk 1 above)
Comment (petr-hajek) [reply src/fetch-url.js]: Changing that line fixed it, but not fully, since there were still errors occuring - for example in the map constructor. I fixed it differently

**Comment by petr-hajek** [comment]
Sorry it took me so long.

I moved the url normalization to a separate file and apply it in the GlMap component as well. It now works with Maplibre and Mapbox renderer as well.

File: src/components/GlMap.svelte [hunk 2]
```diff
@@ -50,7 +51,9 @@
   let mapViewProps = {};
 
   // We can set style (an object) here because mapStyle only changes when it needs to
-  $: ({ style, url } = mapStyle);
+  $: ({ style, url: baseUrl } = mapStyle);
+  $: url = normalizeMapTilerUrl(baseUrl, $configStore.maptilerApiKey);
```
Comment (aparlato) [inline src/components/GlMap.svelte:55]: Hmm, this works, but feels like it isn't totally following existing patterns. By adding a Maptiler key property in our config, we're treating these URL's more similar to a Mapbox URL despite the fact it doesn't have a custom Maptiler protocol (in the way Mapbox has a custom protocol). In theory, if we were to treat Mapbox URLs like this, we would normalize all Mapbox URL's here as well and allow them to also be rendered by Maplibre or Maptiler.

I'm debating whether to:
- move forward with this with a comment to rethink how this pattern works
- apply the same logic to Mapbox URL's to allow them to run on different renderers
- or force the user to either use the Maptiler SDK with this pattern or use the URL with key as a param when wanting to use other renderers.

@ebrelsford can you chime in here with your thoughts on the pattern? Let me know if my explanation here isn't clear enough.

Comment (ebrelsford) [reply src/components/GlMap.svelte:55]: I agree that we should be consistent--if we normalize one provider URL here we should do all the normalizing here too. I'm less concerned about the custom protocol than the fact that each provider does have some amount of work we'll want to do to use their URLs. I think doing this consistently so that Mapbox URLs could be used in other renderers, too, makes sense to me.

I'm not sure that it should happen in this component, I'd be curious what you think about moving up a level so that any necessary changes to a map style all happen up in `Map`, whether it's a `GlMap` or another type?

Comment (petr-hajek) [reply src/components/GlMap.svelte:55]: I agree this would be better to be handled consistently. On the other hand I am a) not really a Svelte expert and b) not so well versed in this library as you are. So I'd prefer to leave this refactoring to someone else if possible.

**Review by aparlato — APPROVED** [review]
Approving with a followup issue I will crosslink
